app/models/label/LabelTable.scala

```diff
@@ -638,6 +655,283 @@ object LabelTable {
+   */
+  def retrieveAssortedLabels(n: Int, loadedLabelIds: Set[Int], severity: Option[Set[Int]] = None): Seq[LabelValidationMetadata] = db.withSession { implicit session => 
+    // List to return.
+    val selectedLabels: ListBuffer[LabelValidationMetadata] = new ListBuffer[LabelValidationMetadata]()
+    Logger.debug("Grabbing random assortment of labels");
+
+    // Init random function
+    val rand = SimpleFunction.nullary[Double]("random")
+
+    // Grab labels and associated information if severity and tags satisfy query conditions.
+    val _labelsUnfiltered = for {
+        _lb <- labelsWithoutDeletedOrOnboarding if _lb.streetEdgeId =!= tutorialStreetId
+        _lt <- labelTypes if _lb.labelTypeId === _lt.labelTypeId
+        _lp <- labelPoints if _lb.labelId === _lp.labelId
+        _ls <- severities if _lb.labelId === _ls.labelId
+        _a <- auditTasks if _lb.auditTaskId === _a.auditTaskId && _a.streetEdgeId =!= tutorialStreetId
+    } yield (_lb, _lp, _lt.labelType, _ls.severity.?)
+    
+    // If severities are specified, filter by whether a label has a valid severity.
+    val _labelsUnfilteredWithSeverity = severity match {
+      case Some(severity) => if (severity.isEmpty) _labelsUnfiltered.filter(_._4 inSet severity)
+                             else _labelsUnfiltered
+      case _ => _labelsUnfiltered
+    }
+
+    // Filter out labels already grabbed before.
+    val _labels = _labelsUnfilteredWithSeverity.filter(label => !(label._1.labelId inSet loadedLabelIds))
+
+    // Join with temporariness to add temporariness attribute.
+    val addTemporariness = for {
+      (l, t) <- _labels.leftJoin(temporariness).on(_._1.labelId === _.labelId)
+    } yield (l._1, l._2, l._3, l._4, t.temporary.?.getOrElse(false))
+
+    // Join with gsvData to add gsv data.
+    val addGSVData = for {
+      (l, e) <- addTemporariness.leftJoin(gsvData).on(_._1.gsvPanoramaId === _.gsvPanoramaId)
+    } yield (l._1, l._2, l._3, l._4, l._5, e.expired)
+
+    // Remove labels with expired panos.
+    val removeExpiredPanos = addGSVData.filter(_._6 === false)
+
+    // Join with descriptions to add descriptions.
+    val addDescriptions = for {
+      (l, d) <- removeExpiredPanos.leftJoin(descriptions).on(_._1.labelId === _.labelId)
+    } yield (l._1.labelId, l._3, l._1.gsvPanoramaId, l._2.heading, l._2.pitch,
+             l._2.zoom, l._2.canvasX, l._2.canvasY, l._2.canvasWidth, l._2.canvasHeight, l._4, l._5, d.description.?)
+
+    Logger.debug(addDescriptions.list.size + " addDescriptions")
+
+    // Randomize and convert to LabelValidationMetadataWithoutTags.
+    val newRandomLabelsList = addDescriptions.sortBy(x => rand).list.map(l => LabelValidationMetadataWithoutTags.tupled(l))
+
+    Logger.debug("got past randomization")
+
+    // TODO: find a place to put this that is more reasonable
+    val labelTypesAsStrings = Set("CurbRamp", "NoCurbRamp", "Obstacle", "SurfaceProblem", "Other", "Occlusion", "NoSidewalk", "Problem")
```

Comment:
I would put this in `LabelTypeTable.scala` as well. I would actually start with this, and then use it to query for the label type IDs. I'm assuming you don't want to include the Problem type anymore, so I'm thinking something like:
```
val validLabelTypes = Set("CurbRamp", "NoCurbRamp", "Obstacle", "SurfaceProblem", "Other", "Occlusion", "NoSidewalk")
val validLabelTypeIds = labelTypes.filter(_.labelType inSet validLabelTypes)
```

Reply:
I understand what you mean by moving it `LabelTypeTable.scala`, but I don't get what you mean by the second part. Are these two vals both supposed to be placed at the top of `LabelTypeTable.scala`?

Reply:
Yes, both should be at the top of LabelTypeTable.scala. I like using the actual strings to get the IDs since that is more understandable and less error prone than also making a list of IDs by hand.

Reply:
Ah gotcha.